rtsp_to_hls/lib/pipeline.ex

```diff
@@ -33,104 +36,91 @@ defmodule Membrane.Demo.RtspToHls.Pipeline do
         }
-      }
-    }
-
-    links = [
-      link(:app_source)
-      |> via_in(Pad.ref(:rtp_input, make_ref()))
-      |> to(:rtp)
+      )
+      |> via_in(:rtp_input)
+      |> child(
+        :rtp,
+        %Membrane.RTP.SessionBin{
+          fmt_mapping: %{96 => {:H264, 90_000}}
+        }
+      ),
+
+      child(
+        :hls,
+        %Membrane.HTTPAdaptiveStream.SinkBin{
+          target_window_duration: Membrane.Time.seconds(120),
+          manifest_module: Membrane.HTTPAdaptiveStream.HLS,
+          storage: %Membrane.HTTPAdaptiveStream.Storages.FileStorage{
+            directory: state[:output_path]
+          }
+        }
+      ),
     ]
 
-    spec = %ParentSpec{children: children, links: links}
-    {{:ok, spec: spec}, %{state | video: %{sps: options[:sps], pps: options[:pps]}}}
+    {[spec: structure], %{state | video: %{sps: options[:sps], pps: options[:pps]}}}
   end
 
   @impl true
-  def handle_notification({:new_rtp_stream, ssrc, 96, _extensions}, :rtp, _ctx, state) do
+  def handle_child_notification({:new_rtp_stream, ssrc, 96, _extensions}, :rtp, _ctx, state) do
     Logger.debug(":new_rtp_stream")
 
-    children = %{
-      video_nal_parser: %Membrane.H264.FFmpeg.Parser{
-        sps: state.video.sps,
-        pps: state.video.pps,
-        skip_until_keyframe?: true,
-        framerate: {30, 1},
-        alignment: :au,
-        attach_nalus?: true
-      },
-      video_payloader: Membrane.MP4.Payloader.H264,
-      video_cmaf_muxer: Membrane.MP4.Muxer.CMAF
-    }
-
-    links = [
-      link(:rtp)
+    structure = [
+      get_child(:rtp)
       |> via_out(Pad.ref(:output, ssrc),
-        options: [depayloader: Membrane.RTP.H264.Depayloader]
+           options: [depayloader: Membrane.RTP.H264.Depayloader])
+      |> child(
+        :video_nal_parser,
+        %Membrane.H264.Parser{
+          sps: state.video.sps,
+          pps: state.video.pps,
+          framerate: {30, 1},
+        }
       )
-      |> to(:video_nal_parser)
-      |> to(:video_payloader)
-      |> to(:video_cmaf_muxer)
-      |> via_in(:input)
-      |> to(:hls)
+      |> via_in(:input, options: [encoding: :H264, segment_duration: Membrane.Time.seconds(4)])
+      |> get_child(:hls)
     ]
 
-    spec = %ParentSpec{children: children, links: links}
-
     actions =
       if Map.has_key?(state, :rtp_started) do
         []
       else
-        [spec: spec]
+        [spec: structure]
       end
 
-    {{:ok, actions}, Map.put(state, :rtp_started, true)}
+    {actions, Map.put(state, :rtp_started, true)}
   end
 
   @impl true
-  def handle_notification({:new_rtp_stream, ssrc, _payload_type, _list}, :rtp, _ctx, state) do
+  def handle_child_notification({:new_rtp_stream, ssrc, _payload_type, _list}, :rtp, _ctx, state) do
     Logger.warn("new_rtp_stream Unsupported stream connected")
 
-    children = [
-      {{:fake_sink, ssrc}, Membrane.Element.Fake.Sink.Buffers}
-    ]
-
-    links = [
-      link(:rtp)
+    structure = [
```

Comment:
[NIT] since there is only a single "branch" in the structure, you don't need to wrap it into a list